bones/dateBone.py

```diff
@@ -225,12 +225,19 @@ def readLocalized(self, value ):
 			res = utc.normalize( res.astimezone( utc ) )
 		return( res )
 
-	def serialize( self, valuesCache, name, entity ):
+	def serialize(self, valuesCache, name, entity):
 		res = valuesCache.get(name)
 		if res:
 			res = self.readLocalized( datetime.now().strptime( res.strftime( "%d.%m.%Y %H:%M:%S" ), "%d.%m.%Y %H:%M:%S"  ) )
+
+			# Crop unwanted values to zero
+			if not self.time:
+				res = res.replace(hour=0, minute=0, second=0, microsecond=0)
+			elif not self.date:
+				res = res.replace(year=0, month=0, day=0)
```

Comment:
This won't work. See https://docs.python.org/3.4/library/datetime.html#datetime.MINYEAR
Also, dates before 1970 cannot be used on all plattforms. So we might want to use the first jan. of 1970?
(Btw: month and day zero are also invalid; it's 1-indexed in this case :)

Reply:
Hi, I've patched your recommendation in the latest commits.

It runs without any problems with the following test-skeleton given:
```python
class testSkel(Skeleton):
    date = dateBone(descr=u"Date", indexed=True)
    dateonly = dateBone(descr=u"Date only", time=False, indexed=True)
    timeonly = dateBone(descr=u"Time only", date=False, indexed=True)
```